### PR TITLE
Numpy backend

### PR DESCRIPTION
**Pull request by ieaves**
## Summary
This PR adds complete numpy backend support for the StandardSet of types. The type implementation is fully compatible with the pandas equivalent implementations with the exception of the `object` type. 

## Caveats

Whereas pandas provides support for `Optional[int]` and `Optional[bool]` numpy doesn't - in order to support those types by default I was forced to make `object` completely disjoint from any other concrete type. A similar story plays out for `datetime` objects with timezones which also default to object in numpy.

### CODE REVIEW

**Comment by ieaves** [comment]
@sbrugman could you give this a review?

File: src/visions/__init__.py [hunk 1]
```diff
@@ -3,12 +3,8 @@
 from visions import types, typesets, utils
 from visions.backends import *
 from visions.declarative import create_type
-from visions.functional import (
-    cast_to_detected,
-    cast_to_inferred,
-    detect_type,
-    infer_type,
-)
+from visions.functional import (cast_to_detected, cast_to_inferred,
```
Comment (sbrugman) [inline src/visions/__init__.py:6]: Seems something going wrong with isort/black

File: src/visions/backends/shared/parallelization_engines.py [hunk 2]
```diff
@@ -45,8 +41,6 @@ def setup(cls, *args, **kwargs) -> None:
         if cls._is_setup:
             return
 
-        import swifter
```
Comment (sbrugman) [inline src/visions/backends/shared/parallelization_engines.py:48]: Any reason to disable swifter in it's entirety?

File: src/visions/__init__.py (hunk 1 above)
Comment (sbrugman) [reply src/visions/__init__.py:6]: Ah, pre-commit is not in the CI checks. Running it will reduce the size of the diff

**Comment by sbrugman** [comment]
Code looks good, will have a try when I have the time for the full review

File: src/visions/backends/shared/parallelization_engines.py (hunk 2 above)
Comment (ieaves) [reply src/visions/backends/shared/parallelization_engines.py]: Mistake on my end. It's a pity this PR is now so old but IIRC I was trying to see if I could roll any numba improvements in as an engine. Should be fixed now.

**Comment by ieaves** [comment]
Final approval @sbrugman ?

**Review by sbrugman — APPROVED** [review]

**Pull request merged**